### PR TITLE
feat: create phone number component  tckt-363

### PR DESCRIPTION
**Pull request by kalasgarov**
Created Phone Number Pattern for Forms Users.
Created Phone Number Pattern Edit for Form Builders.
Added unit tests for Phone Number config file.

### CODE REVIEW

**Review by danielnaab — CHANGES_REQUESTED** [review]
I just have a couple comments, but this looks great.

File: packages/design/src/Form/components/PhoneNumber/PhoneNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          {label}
+          {required && <span className="required-indicator">*</span>}
+        </label>
+        {hint && (
+          <div className="usa-hint" id="primaryPnHint">
+            {hint}
+          </div>
+        )}
+        <input
+          className="usa-input margin-bottom-1"
```
[the remark this answers is not in the document]
Comment (kalasgarov) [reply packages/design/src/Form/components/PhoneNumber/PhoneNumber.tsx:29]: Updated!

File: packages/forms/src/error.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
-export type FormError = {
-  type: FormErrorType;
-  message?: string;
-};
+export type FormError =
+  | {
+      type?: FormErrorType;
+      message?: string;
+    }
+  | undefined;
```
[the remark this answers is not in the document]
Comment (kalasgarov) [reply packages/forms/src/error.ts]: Updated!

**Review by danielnaab — APPROVED** [review]
Approved. I noticed an error class missing from an input (to fix before merging) but otherwise looks great.

File: packages/design/src/FormManager/FormEdit/components/PhoneNumberPatternEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          })}
+        >
+          {message.patterns.phoneNumber.hintLabel}
+          {hint.error ? (
+            <span className="usa-error-message" role="alert">
+              {hint.error.message}
+            </span>
+          ) : null}
+          <input
+            className="usa-input"
```
[the remark this answers is not in the document]
Comment (kalasgarov) [reply packages/design/src/FormManager/FormEdit/components/PhoneNumberPatternEdit.tsx]: Updated!

**Pull request merged**